Review and fix TODO code snippet references (docs 2)

tompratt-AQ opened the pull request:
# Title
Review and fix all TODO code snippet references.  (\docs Part 2)

## Summary
This PR covers the rest of the conceptual files (\docs) that are listed in the dotnet/docs Project "TODO: review snippet reference". It also fixes issues with .config files from the previous #1982 by calling them with an XML code tag. 

## Details
Where there are two identical config files, the one in the language folder (and common subfolder) is used instead of the common folder.

## Suggested Reviewers
@mairaw

tompratt-AQ committed:
Review and fix TODO code snippet references (docs 2)

tompratt-AQ commented:
The following comments are for the changes in this PR:
1. The file docs/framework/wcf/feature-details/how-to-host-a-wcf-service-in-iis.md has an inline xml snippet that is the same as the xml .config snippet file that is being called. Should the inline code be replaced?
2. I added code identifiers to several inline code snippets in some of the files. 
3. I changed snippet tags from csharp style comments to XML style comments in some config files. Since we are now calling them with an XML code tag, I was worried it might not find them.

tompratt-AQ committed:
Fix build error

rpetrusha commented:
@tompratt-AQ, to respond to your question in #1, if there's a choice between incline code and external code, it's best to use external code. (It's more maintainable.)

rpetrusha commented on samples/snippets/csharp/VS_Snippets_CFX/c_standalonedatacontractserializer/cs/source.cs:
```diff
@@ -42,19 +42,19 @@ public class LibraryPatron
         public class LibraryItem
         {
-            //code not shown�
+            //code not shown�
```
What is the final character (rendered as a question mark in a triangle) supposed to be? Make sure the file is saved using UTF-8 encoding (it's probably saved using ASCII encoding).

rpetrusha reviewed (approved):
LGTM, except for one comment about the final character in one C# source code file. And, in response to my last comment, replacing inline code with an external code reference.

tompratt-AQ committed:
Add snippet files and address review comments

tompratt-AQ commented:
I addressed Ron's comments, fixed 4 more files, and fixed one file by re-adding snippets. 

Once these clear, there is only one file left in \docs with "<!-- TODO: review snippet reference" in it:
docs/standard/base-types/how-to-display-localized-date-and-time-information-to-web-users.md which has a snippet tag that links to an .aspx file. Before filing a bug on the issue, it seems like there should be a way to make it work with our current snippet tag system.

tompratt-AQ commented:
Filed issue [982841](https://mseng.visualstudio.com/VSChina/_workitems?id=982841) about the code snippet that links to the .aspx files. So, we'll leave that TODO as the last one in docs. If everything else  checks out here, this should be ready to go!

mairaw committed:
Merge branch 'master' into Review-Snippet-References-Docs-2

mairaw commented on docs/framework/wcf/creating-ws-i-basic-profile-1-1-interoperable-services.md:
```diff
@@ -41,8 +41,8 @@ To configure a [!INCLUDE[indigo2](../../../includes/indigo2-md.md)] service endp
- [!code-vb[C_HowTo-WCFServiceAndASMXClient#0](../../../samples/snippets/visualbasic/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/vb/program.vb#0)]  
- [!code-xml[C_HowTo-WCFServiceAndASMXClient#1](../../../samples/snippets/common/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/common/app.config#1)]
+ [!code-vb[C_HowTo-WCFServiceAndASMXClient#0](../../../samples/snippets/visualbasic/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/vb/program.vb#0)] 
+ [!code-xml[C_HowTo-WCFServiceAndASMXClient#1](../../../samples/snippets/csharp/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/common/app.config#1)]        
```
I'd say as we start changing the references from common to csharp that we also pull the snippet out at the same time and also make sure no other topic is referencing that file.

mairaw commented on docs/framework/wcf/feature-details/config-wcf-service-with-aspnet-web-service.md:
```diff
@@ -47,7 +47,8 @@ To configure a [!INCLUDE[indigo1](../../../../includes/indigo1-md.md)] service e
  [!code-vb[C_HowTo-WCFServiceAndASMXClient#0](../../../../samples/snippets/visualbasic/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/vb/program.vb#0)] 
- <!-- TODO: review snippet reference [!code[C_HowTo-WCFServiceAndASMXClient#1](../../../../samples/snippets/common/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/common/app.config#1)] -->    
+ [!code-xml[C_HowTo-WCFServiceAndASMXClient#1](../../../../samples/snippets/csharp/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/common/app.config#1)] 
+ [!code-xml[C_HowTo-WCFServiceAndASMXClient#1](../../../../samples/snippets/visualbasic/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/common/app.config#1)]   
```
remove this duplicate snippet.

mairaw commented on samples/snippets/common/VS_Snippets_CFX/c_howto_configureclientbinding/common/client.exe.config:
```diff
@@ -1,5 +1,5 @@
-//<snippet100>
-<?xml version="1.0" encoding="utf-8" ?>
+<?xml version="1.0" encoding="utf-8" ?>   
+<!-- <snippet100> -->
```
shouldn't this come before as in the original?

mairaw commented on samples/snippets/common/VS_Snippets_CFX/c_howto_createreliablesessionhttps/common/app.config:
```diff
@@ -1,7 +1,7 @@
+<!-- This is the config for the topic: How to: Create a Custom Reliable Session Binding with HTTPS -->
 
 <?xml version="1.0" encoding="utf-8" ?>
+<!-- <snippet2211> -->
```
same here and in some of the samples below

mairaw reviewed (changes requested):
This is great @tompratt-AQ! Left a few comments.

tompratt-AQ committed:
Make changes per review comments and delete duplicate .config files

tompratt-AQ commented:
@mairaw In this last commit, I did the following:
1. Fixed the specific issues that you flagged in your last comments.
2. For all config files that were in TODO links in this PR: I confirmed that there were duplicates between the Common branch and the csharp branch, made sure that any other reference to the Common file were changed, and then deleted the config file in the Common branch.
3. Deleted other config files that were in the same folder as the deleted dupes after confirming that they also were dupes and weren't being referenced.

mairaw reviewed (approved):
It's great to get these fixes in. Thanks @tompratt-AQ!
